packages/cli/src/admin-sdk.ts

```diff
@@ -0,0 +1,224 @@
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${newSpacePrivateKey.key}`,
+          },
+        }).then(r => r.json());
+      },
+    });
+
+    const spaceModelIdsMap = (Object.values(spaceSettings.cloneInfo.modelIdMap) as string[]).reduce<
+      Record<string, string>
+    >(
+      (modelMap, id) => ({
+        ...modelMap,
+        [id]: createHash('sha256')
+          .update(id + organization.id)
+          .digest('hex'),
+      }),
+      {}
+    );
+    const spaceContentIdsMap = (Object.values(
+      spaceSettings.cloneInfo.contentIdMap
+    ) as string[]).reduce<Record<string, string>>(
+      (contenIdMap, id) => ({
+        ...contenIdMap,
+        [id]: createHash('sha256')
+          .update(id + organization.id)
+          .digest('hex'),
+      }),
+      {}
+    );
+    const replaceIds = (obj: any) =>
+      traverse(obj).map(function (field) {
+        // we keep meta props as is for debugging puprposes
+        if (this.key?.includes('@')) {
+          return;
+        }
+        if (spaceModelIdsMap[field]) {
+          this.update(spaceModelIdsMap[field]);
+        } else if (spaceContentIdsMap[field]) {
+          this.update(spaceContentIdsMap[field]);
+        }
+      });
+
+    const models = await getDirectories(`${directory}`);
+    const modelsPromises = models.map(async ({ name: modelName }) => {
+      const body = replaceField(
+        await readAsJson(`${directory}/${modelName}/schema.model.json`),
+        organization.id,
+        spaceSettings.id
+      );
+      const model = await newSpaceAdminClient.chain.mutation
+        .addModel({ body: replaceIds(body) })
+        .execute({ id: true, name: true });
+      if (model) {
+        const content = (await getFiles(`${directory}/${modelName}`)).filter(
+          file => file.name !== 'schema.model.json'
+        );
+        const modelProgress = multibar.create(content.length, 0, { name: modelName });
+        if (content.length > 0) {
+          modelProgress.start(content.length, 0, { name: modelName });
+        }
+        const writeApi = `https://builder.io/api/v1/write/${modelName}`;
+        const contentPromises = content.map(async (contentFile, index) => {
+          let contentJSON = replaceIds(
+            replaceField(
+              await readAsJson(`${directory}/${modelName}/${contentFile.name}`),
+              organization.id,
+              spaceSettings.id
+            )
+          );
+          modelProgress.increment(1, {
+            name: `${modelName}: writing ${contentFile.name}`,
+          });
+          // post content to write api using the new space private api key
+          await fetch(writeApi, {
+            method: 'POST',
+            body: JSON.stringify(contentJSON),
+            headers: {
+              'Content-Type': 'application/json',
+              Authorization: `Bearer ${newSpacePrivateKey.key}`,
+            },
+          });
```

Comment:
While this is being used only once at the moment I would suggest to also abstract away this into a function (taking just the key and JSON body as params), so future code would just use that function, instead of copy and pasting this block.

Also keep in mind that `fetch` doesn't throw on non-OK HTTP statuses, we would have to check the response's `statusCode` and `throw` accordingly to signal errors during writes to this API.